students/Gabe_Lamberth/lesson04/mailroom2.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+
+# Global data structure
+patrons = {'Emma': [50.00, 150.00, 2000.00, 3000.00], 'Carolyn': [5000.25, 2500.00], 'Andres': [5015.00, 200.50, 35.50]}
+
+# Called when user wants to see the list of donars
+def display_donars():
+    print('Showing list of patrons in the database:')
+    for key, value in patrons.items():
+        print(key, end=', ')
+
+
+# Called from menu_dictionary in main()
+def generate_report():
+    header = ('Donor Name', 'Total Given', 'Num Gifts', 'Average Gift')
+    print('{:25} |{:^15} |{:^15} |{:^15}'.format(*header))
+    print('-' * 75)
+    for key, value in patrons.items():
+        print(f"{key:<30} ${sum(value):^,.2f}  {len(value):^20}  ${sum(value)/len(value):,.2f}")
+
+    print('-- Report Finished --\n')
+
+
+# Called from send_thanks function
+def send_email(choice, amount):
+    print(f"Thank you {choice} for your generous ${amount:,.2f} donation")
+    print('-- Email Sent --\n')
+
+
+# Called from menu_dictionary
+def add_donar():
+    while True:
+        choice = input("Type a patrons name or 'view' to see donor list.\n"
+                       "Type 'quit- to exit to main menu.\n"
+                       "Enter value here >>  ")
+        if choice.lower() == 'view':
+            print('Showing list of patrons in the database:')
+            display_donars()
+        elif choice.lower() == 'quit':
+            break
+        elif choice not in patrons:
+            patrons[choice] = []
+            print(f'{choice} was added to the database')
+            amount = input('Please enter a donation amount: ')
+            patrons[choice].append(float(amount))
+            send_email(choice, float(amount))
+        else:
+            selection = input(f'{choice} already found in database, add another donation? \n Enter Yes or No>> ')
+            if selection.lower() == 'yes':
+                amount = input('Please enter a donation amount: ')
+                patrons[choice].append(float(amount))
+                send_email(choice, float(amount))
+            else:
+                continue
```

Comment:
no need for the `else: continue` here,  it will go to the top of the `while` loop anyway